modules/core/src/main/scala/org/tessellation/infrastructure/trust/storage/TrustStorage.scala

```diff
@@ -6,47 +6,162 @@ import cats.syntax.flatMap._
+
+  case class OrdinalTrustMap(
+    ordinal: SnapshotOrdinal,
+    peerLabels: PublicTrustMap,
+    trust: TrustMap
+  )
+
+  object OrdinalTrustMap {
+    def empty: OrdinalTrustMap = OrdinalTrustMap(SnapshotOrdinal.MinValue, PublicTrustMap.empty, TrustMap.empty)
+
+    def initial: OrdinalTrustMap = OrdinalTrustMap(SnapshotOrdinal(snapshotInterval), PublicTrustMap.empty, TrustMap.empty)
+
+    def advanceInterval(trust: OrdinalTrustMap): OrdinalTrustMap = OrdinalTrustMap(
+      trust.ordinal.plus(snapshotInterval),
+      PublicTrustMap.empty,
+      TrustMap.empty
+    )
+
+    def max(t1: OrdinalTrustMap, t2: OrdinalTrustMap): OrdinalTrustMap =
+      if (t1.ordinal < t2.ordinal) t2 else t1
+  }
 
-  def make[F[_]: Monad: Ref.Make](trustUpdates: Option[PeerObservationAdjustmentUpdateBatch]): F[TrustStorage[F]] =
-    trustUpdates.fold(make)(trust => make.flatTap(_.updateTrust(trust)))
+  def make[F[_]: Monad: Ref.Make]: F[TrustStorage[F]] =
+    Ref[F]
+      .of[TrustStore](TrustStore(TrustMap.empty, OrdinalTrustMap.empty, OrdinalTrustMap.initial))
+      .map(make(_))
 
-  def make[F[_]: Monad](trust: Ref[F, Map[PeerId, TrustInfo]]): TrustStorage[F] =
+  def make[F[_]: Monad: Ref.Make](
+    trustInfo: TrustMap
+  ): F[TrustStorage[F]] =
+    Ref[F]
+      .of[TrustStore](TrustStore(trustInfo, OrdinalTrustMap.empty, OrdinalTrustMap.initial))
+      .map(make(_))
+
+  def make[F[_]: Monad: Ref.Make](
+    trustUpdates: Option[PeerObservationAdjustmentUpdateBatch]
+  ): F[TrustStorage[F]] =
+    trustUpdates.fold(make)(batch => make.flatTap(_.updateTrust(batch)))
+
+  def make[F[_]: Monad](trustStoreRef: Ref[F, TrustStore]): TrustStorage[F] =
     new TrustStorage[F] {
 
-      def updateTrust(trustUpdates: PeerObservationAdjustmentUpdateBatch): F[Unit] =
-        trust.update { t =>
-          t ++ trustUpdates.updates.map { trustUpdate =>
-            trustUpdate.id -> t.getOrElse(trustUpdate.id, TrustInfo()).copy(trustLabel = trustUpdate.trust.value.some)
+      override def updateTrust(trustUpdates: PeerObservationAdjustmentUpdateBatch): F[Unit] =
+        trustStoreRef.update { store =>
+          val t = store.trust
+          val updated = t.trust ++ trustUpdates.updates.map { trustUpdate =>
+            trustUpdate.id -> t.trust.getOrElse(trustUpdate.id, TrustInfo()).copy(trustLabel = trustUpdate.trust.value.some)
           }.toMap
+          store.copy(trust = t.copy(trust = updated))
+        }
+
+      override def updatePredictedTrust(trustUpdates: Map[PeerId, Double]): F[Unit] =
+        trustStoreRef.update { store =>
+          val t = store.trust
+          val updated = t.trust ++ trustUpdates.map {
+            case (k, v) =>
+              k -> t.trust.getOrElse(k, TrustInfo()).copy(predictedTrust = v.some)
+          }
+          store.copy(trust = t.copy(trust = updated))
         }
 
-      def getTrust: F[Map[PeerId, TrustInfo]] = trust.get
+      override def getTrust: F[TrustMap] = trustStoreRef.get.map(_.trust)
```

Comment:
it overrides nothing, so there is no need for `override` modifier here and the rest of methods

Reply:
Removed.